### DISCUSSION TITLE
IEEE address in Open Thread

### DISCUSSION THREAD

**Discussion by jchellap** [openthread/openthread#9342] [Q&A] [answered]
Hi,
The 64-bit IEEE address of the device will be sent by the Host interface for a RCP node or is it something the silicon vendor has to procure a category of address range from the IEEE org? What is the recommendation for IEEE address usage in the Thread?
Basically is it a fixed IEEE address that should be used in the end product that gets stored in the Non volatile or One time memory?

Thanks

**Comment by abtink** [comment, marked as answer]
There are different 64-bit addresses:

1)  The Thread specification requires that the extended MAC address (64 bits) used by a device as the source address in its transmitted 15.4 frames be randomly generated. This same MAC address is used to derive the link-local IPv6 address of the device. After a device successfully attaches to a mesh, this address is persisted in non-volatile settings. This means that if the device is reset, it will continue to use the same MAC address. However, if the device is factory reset, a new random MAC address will be generated and used. In the RCP/host architecture, this address is generated/maintained by host and set on RCP to be used.

2) In addition to this, OpenThread provides API to get a "factory-assigned EUI64" address from device:
https://github.com/openthread/openthread/blob/70adcabca6270eead21f126533c697d4d1637b77/include/openthread/link.h#L592-L599
This address itself is retrieved using platform API `otPlatRadioGetIeeeEui64()`, so it is up to platform (and product integrator) to determine what is returned. Generally, radio chip vendors provide a EUI64 based on their OUI for this. However, platforms/product vendors can choose to override this and return whatever makes sense for their product.
Also note that this address itself is not really used by Thread stack (e.g., in tx frames).  It can be optionally used as part of MeshCop to identify a Joiner.

**Reply by jchellap** [reply]
Thanks for the clarification @abtink!
